### DISCUSSION TITLE
Seeking Installation Instructions & Advice on setting up & hosting a private Linux Bare Metal Server at Home (Ubuntu 22.04 or 24.04)

### DISCUSSION THREAD

**Discussion by 2loki4u** [lllyasviel/Fooocus#3037] [Q&A]
I am looking to build a SSF server at home for all things AI and it seems to me that given its mixed-AI-use, running Linux natively (ubuntu - either 22.04 or 24.04 - still to be decided) on it would likely be better than Windows w/linux VMs. 

My goal is to locally host various private AI tools such as Fooocus, ollama & others. When I was reading through the download landing page for installing Fooocus, I find it lacks substantive information on doing something like this. Any useful and specific resources related to this endeavor would be greatly appreciated, perhaps I've overlooked something.

So, am I pursuing a herculean goal that makes sense or is this a fool's errand? Thoughts appreciated.

In short, anyone having experience with such an endevour, who's willing to share their wisdom/experience/input would be welcomed & IMMENSELY APPRECIATED.

The tl;tr - 
My reason for pursuing a bare metal Linux-based dedicated AI server is that all of our existing machines have integrated AMD or ARC graphics, except my main workstation, which is all AMD but lacks the resources to host this in parallel with its other duties/functions plus I can't get the rx7800xt to be recognized in ubuntu in WSL for any AI applications.

The planned system I've spec'd out, will be configured with: an intel 13gen i7/nvidia rx4070ti-16gb w/64gb ddr5 & pcie4 2tb nvme (sadly, that's the budget I've got to work with).

I'm a novice running Linux so detailed instructions I can reference or links to pages I missed that further explain the setup in Linux systems - specifically Ubuntu and which version is preferred would be incredibly helpful. 

Ultimately, these tools will need to be accessible on our windows machines via our browsers.

Thanks in advance
~Chris

**Comment by Rogala** [comment]
1. Will you be working by yourself in a single thread on your server?
2. Will there be many of you connecting to this server?
3. Will you connect and use its resources at the same time?
4. How will you divide the capacity of the GPU for different programs or different threads from different people?

**Reply by 2loki4u** [reply]
Greetings Rogala, thanks for replying to my inquiry.

1. 2 people typically connected
2. no - just the two of us (this is a home application) yes - but different applications
3. The most common scenario will be my using LLM Chat AI while my other half uses the system for Fooocus Image Rendering and manipulation.
4. This is a big question to which I do not know how to answer. A major concern raised today is the selection of an i7 13700k CPU and how to divide resources if each of the applications were to be virtualized in some way - given that it only has 8 dual-thread performance cores and 8 single-thread efficiency cores. No idea how VM solutions would handle such a processor. Ideally, I would like all of these tools to live in a single instance in linux - better still without virtualization (aside from docker perhaps). This is the area I am most apprehensive about.

**Reply by Rogala** [reply]
I'll answer briefly, it is impossible to share resources of game version GPUs, only server GPUs with purchased drivers know how to distribute power to different applications and virtual machines.

Make a schedule and work one by one.

**Comment by 2loki4u** [comment]
Thank you - this was my assumption and operating premise in general.

If i installed two utilities, for argument's sake - Ollama & Fooocus within a bare metal installation of Ubuntu 24.04, would they both be accessible at the same time? Meaning, what if I did not create a VM for each. Wouldn't it then work just like if I am watching a video on the side on one monitor on my windows machine while running a LLM inquiry via a web chat based bot?

I assume of course that the webUI interfaces of each application will have it's own local host IP that can be exposed (passed through) the machine's firewall and made accessible to the network for remote access.

Just curious...


~Chris
________________________________

**Reply by mashb1t** [reply]
Hey @2loki4u (Chris),

> would they both be accessible at the same time?

Yes, already doing so on my own servers. You can server both tools with --listen (0.0.0.0, catch all) with separate ports, also manageable in your firewall even on the same host. For highest throughput and driver compatibility i'd recommend not to use virtualisation at all, or use Docker containers for all of the services on one V-Host(as you've already mentioned).
Resource sharing will be available by default, but please be aware that also both tools do then use the GPU CUDA cores when generating at the same time, so inference will be slower and depending on your VRAM the models might not be placed with maximum efficiency (swap used when VRAM is full). On servers with 4090 / A6000 I have not encountered any issues with parallel provisioning, also running a music generation service and additional containers there.
For windows firewall i have only those inbound ports configured as well as RDP, so this should also be covered, and i have a [reverse proxy](https://github.com/NginxProxyManager/nginx-proxy-manager) in front of these services to only allow incoming traffic through the proxy to the machine (allows user management, sharing from internet, user basic auth, etc.).
Hope this helps!

**Reply by Rogala** [reply]
Getting access to the interface is a trivial task, but getting a CUDA error about memory shortage and program crash is very fast.

These tasks are not parallel video watching, the AI will take all available resources in CPU, Memory, GPU, I/O, at once for one running task, the second will wait for a long time and grab pieces of available resources. 

As a result, your PC may simply freeze and you will lose some of your work.

4090 and A6000 are a bit redundant for Fooocus, so it's easier for them to give resources to another application, but 4070ti is not sure if it can handle it or not.

**Reply by mashb1t** [reply]
quick maths:
- Ubuntu: 0.5GB VRAM (if used in desktop mode, can highly recommend Debian)
- Fooocus: 8-10GB VRAM (peak when generating, normally about 5-6GB) img2img higher)
- ollama with 7b LLM (4 bit quantized): 7GB VRAM

=> ~15-17 GB VRAM needed, will outsiurce to RAM/swap if maxed out

**Reply by 2loki4u** [reply]
thank you for those details. that's good to know, I can make adjustments to the build and set reasonable expectations based on that.
